synapse/handlers/federation_event.py

```diff
@@ -1031,6 +1031,12 @@ async def _get_state_ids_after_missing_prev_event(
             InvalidResponseError: if the remote homeserver's response contains fields
                 of the wrong type.
         """
+
+        # It would be better if we could query the difference from our known
+        # state to the given `event_id` so the sending server doesn't have to
+        # send as much and we don't have to process as many events. For example
+        # in a room like #matrixhq, we get 200k events (77k state_events, 122k
+        # auth_events) from this call.
```

Comment:
> no objection to adding a comment here, but it feels like it's just going to sit here and never be heard from again.
> 
> Is it worth instead having an issue or a spec issue about this? You could then link to it from the comment, even.
>
> *-- https://github.com/matrix-org/synapse/pull/13586#pullrequestreview-1081874011*

I would like to create a MSC around this.

But perhaps an issue will suffice for now ⏩

Reply:
Created https://github.com/matrix-org/synapse/issues/13618